04.ls/ls.rb

```diff
@@ -1,15 +1,96 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
+require 'etc'
+
 require 'optparse'
 
 SPLIT_NUMBER = 3
 
+FILE_TYPE = {
+  '01' => 'p',
+  '02' => 'c',
+  '04' => 'd',
+  '06' => 'b',
+  '10' => '-',
+  '12' => 'l',
+  '14' => 's'
+}.freeze
+
+FILE_MODE = {
+  '0' => '---',
+  '1' => '--x',
+  '3' => '-w-',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
 def main
-  options = ARGV.getopts('r')
+  options = ARGV.getopts('l')
   files = Dir.glob('*')
-  sorted_files = options['r'] ? files.reverse : files
-  display_files(sorted_files)
+  options['l'] ? display_long_format(files) : display_files(files)
+end
+
+def display_long_format(files)
+  long_formats = build_detailed_files(files)
+  total_block_number = calc_total_block_number(long_formats)
+  max_width = calc_max_width(long_formats)
+
+  puts "total #{total_block_number}"
+  long_formats.each do |format|
+    print "#{format[:file_mode]}  "
+    print "#{format[:hard_links_count].to_s.rjust(max_width[:link])} "
+    print "#{format[:owner].ljust(max_width[:owner])}  "
+    print "#{format[:group].ljust(max_width[:group])}  "
+    print "#{format[:file_size].to_s.rjust(max_width[:file_size])} "
+    print "#{format[:day_time]} "
+    print "#{format[:file_name]}\n"
+  end
+end
+
+def build_detailed_files(files)
+  files.map do |file|
+    file_info = File.stat(file)
```

Comment:
```suggestion
    file_stat = File.stat(file)
```

の方が良さそうです。

https://bootcamp.fjord.jp/pages/readable-variable-name